# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/storage/SessionStorageCache.h
# Commit: aea23435804b
# Full Hash: aea23435804b56798dbae8fa46dfccb8cbdca131
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-12-13 16:53:24
# Regressor Bug: 1593246
# File Overlap Count: 6
# Description:
#   Bug 1593246 - Part 5: Replicate session storage data r=janv,asuth
#   
#   This commit implements a simple data replication scheme, where
#   each content process receives a copy of the relevant session
#   storage data when navigating, and sending all its session storage
# ==============================================================================

diff -r da1672f831d1 -r aea23435804b dom/storage/SessionStorageCache.h
--- a/dom/storage/SessionStorageCache.h	Fri Dec 13 09:16:32 2019 +0000
+++ b/dom/storage/SessionStorageCache.h	Fri Dec 13 09:45:55 2019 +0000
@@ -12,6 +12,8 @@
 namespace mozilla {
 namespace dom {
 
+class KeyValuePair;
+
 class SessionStorageCache final {
  public:
   NS_INLINE_DECL_REFCOUNTING(SessionStorageCache)
@@ -44,6 +46,10 @@
 
   already_AddRefed<SessionStorageCache> Clone() const;
 
+  nsTArray<KeyValuePair> SerializeData(DataSetType aDataSetType);
+  void DeserializeData(DataSetType aDataSetType,
+                       const nsTArray<KeyValuePair>& aData);
+
  private:
   ~SessionStorageCache() = default;
 